packages/esphome/desk_height_sensor.h

```diff
@@ -138,16 +141,20 @@ class DeskHeightSensor : public Component, public UARTDevice, public Sensor
           int height1 = hex_to_int(history[1]) * 100;
           int height2 = hex_to_int(history[0]) * 10;
           int height3 = hex_to_int(incomingByte);
-
-          float finalHeight = height1 + height2 + height3;
-
-          if (is_decimal(history[0]))
+          if (height2 == 100) // check if 'number' is a hyphen, return value 10 multiplied by 10
           {
-            finalHeight = finalHeight / 10;
+            value =  lastPublished;
```

Comment:
You don't need to set value to last published. In this case we just don't want to set anything.

Reply:
@mad2xlc did you push this change already? So we can merge it in :D.